src/main/java/org/apache/sling/feature/cpconverter/vltpkg/VaultPackageAssembler.java

```diff
@@ -198,14 +200,18 @@ public File getEntry(String path) {
     }
 
     public void updateDependencies(Map<PackageId, Set<Dependency>> mutableContentsIds) {
+        Map<Dependency, Set<Dependency>> matches = new HashMap<>();
         for (Dependency dependency : dependencies) {
```

Comment:
why not using  https://docs.oracle.com/javase/8/docs/api/java/util/Iterator.html#remove--